src/Model/ParameterGo.cs

```diff
@@ -29,11 +29,22 @@ public ParameterGo()
         public void AddImports(HashSet<string> imports)
         {
             ModelType.AddImports(imports);
-            // we use fmt.Sprintf to format certain header and/or query params, see GetStringFormat in this file
-            if ((Location == Core.Model.ParameterLocation.Header || Location == Core.Model.ParameterLocation.Query) &&
-                !ModelType.IsPrimaryType(KnownPrimaryType.String) && !ModelType.IsDateTimeType())
+            // related to codegen in GetStringFormat() in this file
+            if ((Location == Core.Model.ParameterLocation.Header || Location == Core.Model.ParameterLocation.Query))
             {
-                imports.Add(PrimaryTypeGo.GetImportLine(package: "fmt"));
+                if (ModelType.IsPrimaryType(KnownPrimaryType.Int) || ModelType.IsPrimaryType(KnownPrimaryType.Long) ||
+                    ModelType.IsPrimaryType(KnownPrimaryType.Double))

```

Comment:
Agreed, I really have how we have to duplicate this logic (I would really like to overhaul how we handle the addition of imports to avoid this).

Reply:
Yea, I think we would need an overhaul of a lot of stuff. 😄   . Given how the base types come to us it would be really nice to have virtual extension methods but apparently not going to happen.

Reply:
actually maybe we can just start defining some interfaces with the methods needed and then implement them on the types. and when needed we use the type as that interface ?

Reply:
Here they are all the same type (PrimaryTypeGo).  There is also the context in which we see them, i.e. as members of structs or parameters, and depending on this context the imports to add will be different (in this PR the imports only apply when the type is a parameter).  So I think we would still need to segregate them between these two cases.